### DISCUSSION TITLE
Add UserId to syncEvent

### DISCUSSION THREAD

**Discussion by Justincale** [oqtane/oqtane.framework#4242] [Ideas]
Hi,

I am working on some modules that rely quite heavily on syncEvents and have come across the need to have access to the UserId of the user that was logged in when the syncEvent was created. How do you feel about adding this?

**Comment by sbwalker** [comment]
@Justincale you would need to explain more about how you are using SyncEvents (https://www.oqtane.org/blog/!/53/oqtane-server-events). Are you using them to coordinate processing on the server or client? Adding UserId would be a very invasive change as in order for the value to be consistently available, every call to AddSyncEvent() would need to be modified. This is possible in the framework but there may be third party modules using AddSyncEvent() as well - and in those cases UserId would not be populated (until the module was updated) - which means developers could not 100% rely on this property.

**Reply by Justincale** [reply]
In my use case the addition of UserId would be used to coordinate processing on the server. I envision this to be nullable due to at least one area of Oqtane that creates sync events without an authenticated user (eg: file download), and could be added as an optional parameter to the existing addSyncEvent methods. In this way it would be up to Oqtane/3rd party components to send UserId to addSyncEvent only when needed, and it won’t break existing code.

**Reply by sbwalker** [reply]
@Justincale are you asking for the ability for custom modules to include a UserId parameter... or are you expecting that the core framework would also include the UserId in all of its AddSyncEvent() calls... or both?

**Reply by Justincale** [reply]
Both.

**Reply by Justincale** [reply]
Use Case: I want to send an email to the user who downloads a particular file.
